src/Components/Cards/TextCard/TextCard.jsx

```diff
@@ -117,13 +115,13 @@ export const TextCard = ({
         <div className="post-footer">
           <div className="post-footer-child-left">
             {
-              !liked && <span onClick={likedHandler} className="material-icons post-footer-icons">
+              !userLiked()&& <span onClick={likedHandler} className="material-icons post-footer-icons">
```

Comment:
Instead of repeatedly calling the userLiked() function, calling it once and storing the result in a constant would be better. Similarly for all the places you've called the function